analyzer/evmabibackfill/evm_abi_backfill_test.go

```diff
@@ -0,0 +1,298 @@
+package evmabibackfill
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"testing"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/modules/evm"
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/nexus/analyzer/evmabi"
+	"github.com/oasisprotocol/nexus/common"
+	"github.com/oasisprotocol/nexus/log"
+)
+
+// TODO: Update WROSE source and add tests for more txs, mismatched txs, and custom errors
+
+type mockParsedEvent struct {
+	Name *string
+	Args []*abiEncodedArg
+	Sig  *ethCommon.Hash
+}
+
+type mockParsedTxCall struct {
+	Name *string
+	Args []*abiEncodedArg
+}
+
+func unmarshalEvmEvent(t *testing.T, body []byte) *abiEncodedEvent {
+	var ev evm.Event
+	err := json.Unmarshal(body, &ev)
+	require.Nil(t, err)
+	return &abiEncodedEvent{
+		Round:     0,
+		TxIndex:   nil,
+		EventBody: ev,
+	}
+}
+
+func unmarshalEvmTx(t *testing.T, body string, error_message *string) *abiEncodedTx {
+	txData, err := base64.StdEncoding.DecodeString(body)
+	require.Nil(t, err)
+	return &abiEncodedTx{
+		TxHash:         "", // not relevant for current unit tests
+		TxData:         txData,
+		TxRevertReason: error_message,
+	}
+}
+
+func TestParseEvent(t *testing.T) {
+	abi := evmabi.WROSE
+	p := &processor{
+		runtime: common.RuntimeSapphire,
+		target:  nil,
+		logger:  log.NewDefaultLogger("testing"),
+	}
+	evs := []*abiEncodedEvent{
```

Comment:
Do these test cases cover different code paths inside `p.parseEvent()`? If not, we can prune this down to just one case. The regression tests have more of a more-is-better philosophy; here, it's nice to have fewer tests so that 1) the intent behind each one can be easier to understand and 2) there's less to fix/maintain/update if the code changes.

OTOH if it does make sense to keep multiple test cases, it would be really nice to keep the inputs (= this array) next to the expected outputs. Basically, the `mockParsedEvent` type would gain a new field `Input` or sth. This [short section 10](https://www.practical-go-lessons.com/chap-19-unit-tests#how-to-write-a-table-test) gives a nice example and motivation, and also explains how to use `t.Run()` so we don't get failure messages from `evm_abi_backfill_test:155` (inside the test's final for loop) without knowing which iteration of the loop failed.

Reply:
Got it, pruned down to a `Transfer` event!